### PR TITLE
Fixes godmode puppet bug

### PR DESCRIPTION
**Pull request by Aporhtonoma**
## About The Pull Request
apparently applying warding pheromones AND THEN giving your puppet a warding blessing makes it reach god mode due to the blessing being multiplicative. This changes that to it a being flat bonus and should kill this bug forever
## Why It's Good For The Game
image
this never happens again.
## Changelog
:cl:
fix: fixes puppets ridiculous armor ratings if warding pheromones were applied before it got warding blessing
/:cl:

### CODE REVIEW

**Review by Lumipharon — CHANGES_REQUESTED** [review]
bruh, what is this shitcode? Also you closed the revert pr while this isn't even merged yet.

**Comment by Aporhtonoma** [comment]
> bruh, what is this shitcode? Also you closed the revert pr while this isn't even merged yet.

I have no idea I changed the code as little as I could because I did not wanna break something. Bonus armor is additive and the final values I left unchanged since I dint wanna touch balance

as for the revert pr... Im stupid

**Review by Lumipharon — COMMENTED** [review]
What are these cursed numbers and why do puppets need exactly 105.84 extra melee armour?

**Comment by Aporhtonoma** [comment]
> What are these cursed numbers and why do puppets need exactly 105.84 extra melee armour?

this cursed numbers are the result of (the original armor of the puppet)*aurastrength*2.7
and the 105.84 is 14(melee armor)*2.8(aurastrength)*2.7
Im really doing my best to not make this a balance pr :(

File: code/datums/status_effects/xeno_buffs.dm [hunk 1]
```diff
@@ -831,20 +831,17 @@
 
 /datum/status_effect/blessing/warding
 	id = "blessing of warding"
-	///A holder for the exact armor modified by this status effect
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:834]: ???

File: code/datums/status_effects/xeno_buffs.dm [hunk 2]
```diff
@@ -831,20 +831,17 @@
...
 	var/datum/armor/armor_modifier
 
 /datum/status_effect/blessing/warding/on_apply()
-	buff_owner = owner
-	if(!isxeno(buff_owner))
+	if(!isxeno(owner))
 		return FALSE
-	armor_modifier = buff_owner.soft_armor.scaleAllRatings(strength * 2.7)
-	buff_owner.soft_armor = buff_owner.soft_armor.attachArmor(armor_modifier)
+	armor_modifier = getArmor(14 * strength * 2.7 ,3 * strength * 2.7 ,5 * strength * 2.7 ,3 * strength *2.7 ,0 ,0 , 0 ,0 )//create armor for puppets
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:839]: why are you complicating it with this 2.7 mult, on top of the strength mult?

File: code/datums/status_effects/xeno_buffs.dm [hunk 3]
```diff
@@ -831,20 +831,17 @@
...
+	owner.soft_armor=owner.soft_armor.attachArmor(armor_modifier)//we add armor to the puppet
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:840]: spacing, also kill the pointless comment

File: code/datums/status_effects/xeno_buffs.dm [hunk 4]
```diff
@@ -831,20 +831,17 @@
...
 	return TRUE
 
 /datum/status_effect/blessing/warding/on_remove()
-	buff_owner.soft_armor = buff_owner.soft_armor.detachArmor(armor_modifier)
-	armor_modifier = null
+	owner.soft_armor=owner.soft_armor.detachArmor(armor_modifier)//we take away armor on blessing removal
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:844]: as above

File: code/datums/status_effects/xeno_buffs.dm [hunk 5]
```diff
@@ -831,20 +831,17 @@
...
 		return FALSE
-	armor_modifier = buff_owner.soft_armor.scaleAllRatings(strength * 2.7)
-	buff_owner.soft_armor = buff_owner.soft_armor.attachArmor(armor_modifier)
+	armor_modifier = getArmor(14 * strength * 2.7 ,3 * strength * 2.7 ,5 * strength * 2.7 ,3 * strength *2.7 ,0 ,0 , 0 ,0 )//create armor for puppets
+	owner.soft_armor=owner.soft_armor.attachArmor(armor_modifier)//we add armor to the puppet
 	return TRUE
 
 /datum/status_effect/blessing/warding/on_remove()
-	buff_owner.soft_armor = buff_owner.soft_armor.detachArmor(armor_modifier)
-	armor_modifier = null
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:847]: why'd you remove this? this proc was fine other than referencing owner instead of buff_owner

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/xeno_buffs.dm (hunk 2 above)
Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm:839]: cause this was the original calculation for the armor propably needs removing or tweaking but I dont want to mess with the balance only fix the bug

File: code/datums/status_effects/xeno_buffs.dm (hunk 1 above)
Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm:834]: woops

File: code/datums/status_effects/xeno_buffs.dm (hunk 3 above)
Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm]: done

File: code/datums/status_effects/xeno_buffs.dm (hunk 4 above)
Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm]: done

File: code/datums/status_effects/xeno_buffs.dm (hunk 2 above)
Comment (Lumipharon) [reply code/datums/status_effects/xeno_buffs.dm:840]: this doesn't need to be a separate value, this does literally nothing of any use

Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm]: true I added it directly into the integer value

File: code/datums/status_effects/xeno_buffs.dm [hunk 6]
```diff
@@ -837,7 +837,7 @@
 /datum/status_effect/blessing/warding/on_apply()
 	if(!isxeno(owner))
 		return FALSE
-	armor_modifier = getArmor(37.8 * strength ,8.1 * strength ,13.5 * strength * 2.7 ,8.1 * strength ,0 ,0 ,0 ,0 )//create armor for puppets
+	armor_modifier = getArmor(37.8 * strength ,8.1 * strength ,13.5 * strength ,8.1 * strength ,0 ,0 ,0 ,0 )//create armor for puppets
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:840]: also remove all these zeros, why are you adding extra args?

Comment (Aporhtonoma) [reply code/datums/status_effects/xeno_buffs.dm]: removed and slain.

File: code/datums/status_effects/xeno_buffs.dm [hunk 7]
```diff
@@ -835,15 +835,14 @@
 	var/datum/armor/armor_modifier
 
 /datum/status_effect/blessing/warding/on_apply()
-	buff_owner = owner
-	if(!isxeno(buff_owner))
+	if(!isxeno(owner))
 		return FALSE
-	armor_modifier = buff_owner.soft_armor.scaleAllRatings(strength * 2.7)
-	buff_owner.soft_armor = buff_owner.soft_armor.attachArmor(armor_modifier)
+	armor_modifier = getArmor(37.8 * strength ,8.1 * strength ,13.5 * strength ,8.1 * strength )
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:840]:
```suggestion
	armor_modifier = getArmor(37.8 * strength ,8.1 * strength ,13.5 * strength ,8.1 * strength)
```

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**